src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -986,22 +779,440 @@ public void AnalyzeCommand_LoadConfigurationFile(string configValue, bool defaul
+            public IList<string> Files;
+
+            public IFileSystem FileSystem
+            {
+                get
+                {
+                    if (_fileSystem == null)
+                    {
+                        _fileSystem = CreateDefaultFileSystemForResultsCaching(Files);
+                    }
+                    return _fileSystem;
+                }
+
+                set => _fileSystem = value;
+            }
+
+            public int ExpectedReturnCode;
+
+            public int ExpectedResultsCount =>
+                // Non-verbose results
+                (ExpectedErrorCount + ExpectedWarningCount) +
+                // Verbose results
+                (Verbose
+                    ? ExpectedNoteCount + ExpectedPassCount + ExpectedInformationalCount +
+                      ExpectedOpenCount + ExpectedReviewCount + ExpectedNotApplicableCount
+                    : 0);
+
+            public int ExpectedErrorCount =>
+                Files.Where((f) => f.Contains("Error")).Count() +
+                // For our special case, all files except for those that are marked as 'not applicable' will
+                // produce a 'pdb load' notification that will be converted to an error. The not applicable
+                // cases will not do this, because the return of 'not applicable' from the CanAnalyze
+                // method will result in Analyze not getting called subsequently for those scan targets.
+                (NotificationsWillBeConvertedToErrorResults 
+                    ? Files.Count() - Files.Where((f) => f.Contains("NotApplicable")).Count()
+                    : 0);
+
+            public int ExpectedWarningCount =>
+                Files.Where((f) => f.Contains("Warning")).Count();
+
+            public int ExpectedNoteCount => Verbose
+                ? Files.Where((f) => f.Contains("Note")).Count()
+                : 0;
+
+            public int ExpectedPassCount => Verbose
+                ? Files.Where((f) => f.Contains("Pass")).Count()
+                : 0;
+
+            public int ExpectedInformationalCount => Verbose
+                ? Files.Where((f) => f.Contains("Informational")).Count()
+                : 0;
+
+            public int ExpectedReviewCount => Verbose
+                ? Files.Where((f) => f.Contains("Review")).Count()
+                : 0;
+
+            public int ExpectedOpenCount => Verbose
+                ? Files.Where((f) => f.Contains("Open")).Count()
+                : 0;
+
+            public int ExpectedNotApplicableCount => Verbose
+                ? Files.Where((f) => f.Contains("NotApplicable")).Count()
+                : 0;
+
+            public bool PersistLogFileToDisk;
+
+            public TestRuleBehaviors TestRuleBehaviors;
+
+            // TODO: this is an special knob that that accounts for a current SDK behavior.
+            // Specifically, the MSBuildConverter currently transforms all notifications
+            // to results. This will require us 
```

Comment:
>us [](start = 45, length = 2)

... to do what? :-)

Also, is there a bug tracking this TODO? (If so, mention it here.)